src/relayedTransactionV2Builder.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { loadTestWallets, TestWallet } from "./testutils";
+import { RelayedTransactionV2Builder } from "./relayedTransactionV2Builder";
+import { Address } from "./address";
+import { TransactionPayload } from "./transactionPayload";
+import { assert } from "chai";
+import { Transaction } from "./transaction";
+import * as errors from "./errors";
+
+describe("test relayed v2 transaction builder", function () {
+    let alice: TestWallet, bob: TestWallet;
+
+    before(async function () {
+        ({alice, bob} = await loadTestWallets());
+    });
+
+    it("should throw exception if args were not set", async function () {
+        const builder = new RelayedTransactionV2Builder();
+        assert.throw(() => builder.build(), errors.ErrInvalidRelayedV2BuilderArguments);
+    });
+
+    it("should throw exception if gas limit of the inner tx is not 0", async function () {
+        let builder = new RelayedTransactionV2Builder();
+
+        let networkConfig = {
+            MinGasLimit: 50_000,
+            GasPerDataByte: 1_500,
+            GasPriceModifier: 0.01,
+            ChainID: "T"
+        };
+
+        const innerTx = new Transaction({
+            nonce: 15,
+            receiver: Address.fromBech32("erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u"),
+            gasLimit: 10,
```

Comment:
Using some implementation of `IGasLimit`, such as `{ valueOf: function() { return 10; } }` might fail the test.

Reply:
extended the test to include this as well